src/som/interpreter/actors/Actor.java

```diff
@@ -300,17 +299,7 @@ private void execute(final EventualMessage msg,
         TracingActor.handleBreakpointsAndStepping(msg, dbg, actor);
       }
 
-      try {
-        if (VmSettings.KOMPOS_TRACING) {
-          KomposTrace.scopeStart(DynamicScopeType.TURN, msg.getMessageId(),
-              msg.getTargetSourceSection());
-        }
-        msg.execute();
-      } finally {
-        if (VmSettings.KOMPOS_TRACING) {
-          KomposTrace.scopeEnd(DynamicScopeType.TURN);
```

Comment:
Where does the scopeEnd go? (scopes can be nested, e.g. for transactions)

Reply:
ReceivedRootNode handlees scope start and end. My assumption was that you forgot to remove this code when you did some refactoring.

Reply:
Ok, but then I believe the `scopeStart` you write after swapping buffers is not correct.

`scopeStart` and `scopeEnd` should be balanced. And it should be possible to order the buffers on reading so that `scopeStart` in a previous buffer isn't an issue and doesn't need to be repeated, no?

Reply:
The Idea was that ScopeStart implicitly ends the previous scope. If scopes can be nested we will need a different solution that keeps track of all open scopes, and leaves enough space to close the scopes at the end of the buffer

Reply:
Yeah, scopes are supposed to be nest-able, at least according to [our paper](https://stefan-marr.de/downloads/dls17-marr-et-al-concurrency-agnostic-protocol-for-debugging.pdf). They are used for locks for instance, each `acquire()` opening a new scope, which means, technically, they are not even guaranteed to nest correctly. And they are used for transactions, which can be correctly nested.

Reply:
The Scope End event doesn't contain information on the scope that is ended, isn't that a problem in the case of locking?

Reply:
Yeah, indeed. I suppose the current implementation assumes correct nesting then. Another bug ☹️

Reply:
Ok, I guess we should assume at least support for correct nesting at the moment.

Reply:
The most simple solution would be to have a counter that indicates the number of open scopes, and use that counter when checking remaining space, inserting n endscope events on buffer swap.

Reply:
Why do you need to end the scope?
And how do you distinguish buffer-swap scope closing from explicit scope ends?

Reply:
Hm, not sure about the details. I think there is some current thread info as well, we could store the open scopes when a new thread takes over, and when the original thread appears again restore the context. then we would not need close scope events and have the correct context for all events

Reply:
In the current system, I do not think that scopes move threads. At least not as far as I can recall. So, it should be new buffers on the same thread only. So, ordering them should be possible, and no extra meta data needed, no?

Reply:
Yes, what i described has to be done in the parser

Reply:
Ok, does the `scopeStart` then need to be removed? (cf https://github.com/smarr/SOMns/pull/304/files#r308452408)

Reply:
I think so, the parser in kompos probably needs to be checked that this is handled correctly.